marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java

```diff
@@ -204,16 +201,18 @@ private void applyOptionsToEndpointConstants(ObjectNode endpointConstants, Map<S
         });
     }
 
-    private void loadEndpoints() throws IOException {
-        String scriptPath = "marklogic-data-hub-spark-connector/bulkIngester.sjs";
-        String apiPath = "marklogic-data-hub-spark-connector/bulkIngester.api";
-
-        TextDocumentManager modMgr = hubClient.getModulesClient().newTextDocumentManager();
-        DocumentWriteSet writeSet = modMgr.newWriteSet();
-        DocumentMetadataHandle metadata = buildDocumentMetadata();
-        writeSet.add("/" + apiPath, metadata, new InputStreamHandle(new ClassPathResource(apiPath).getInputStream()).withFormat(Format.JSON));
-        writeSet.add("/" + scriptPath, metadata, new InputStreamHandle(new ClassPathResource(scriptPath).getInputStream()).withFormat(Format.TEXT));
-        modMgr.write(writeSet);
+    private void loadEndpointsIfNotPresent(String scriptPath, Format format) {
```

Comment:
Can you rename this to "loadModuleIfNotPresent" and rename "scriptPath" to "modulePath"?